### DISCUSSION TITLE
Note duration doubling when writing to file

### DISCUSSION THREAD

**Discussion by bgianardo** [mido/mido#425] [Q&A] [answered]
[MIDINegator.zip](https://github.com/mido/mido/files/3906011/MIDINegator.zip)

Hi everybody.
I wanted to write a program that could take a given mid file, apply a note-by-note pitch transformation according to the theory of negative harmony, and  write the result to another mid file.

Basically:

1. read a message from the source file
2. if it is not a note_on or note_off event, rewrite it unchanged
3. otherwise, detect the pitch value, apply a transformation based on the given root note
4. substitute the old pitch value with the new
5. write the new message

When I open the output file in the same DAW I used to compose the source one, I can see the pitch change occurred correctly, but unexplicably all events durations are DOUBLED.

I listed the message from both files and, to my amazement, they are IDENTICAL (save for the pitches, which are correctly modified)

You can find hereby attached the Python code, the source file and the output file.

Does anybody have suggestions about this weird behaviour, or about how to approach the problem in some other way?

Thank you very much
Bruno

**Comment by dsheng3** [comment, marked as answer]
In case somebody else has the same problem again (since it took me quite some to figure out)

You need to set/copy the ticks_per_beat for your Midi-File:
```py
mid_in = MidiFile('some file path')
mid_out = MidiFile(ticks_per_beat=mid_in.ticks_per_beat))
```

In your case your original .mid-File had 960 ticks per beat which is twice of the default 480 ticks per beat.

**Reply by Isostica** [reply]
Thank you for this!  I'm trying to convert some drum midi mappings, and my initial effort had the midioutfile playing at 480 ticks per beat, which was the default when created, but my original drum midi was at 240.  So the newly created midi was playing at double speed. I was trying to figure out how to change the ticks_per_beat for the midioutfile but couldn't figure out how.  I was trying  to set midifile_out.ticks_per_beat =  240 after creation, but couldn't.  This answer is so helpful and works perfectly.  I would've been lost about how to fix without this post.  Thank you  ;-)

**Comment by bgianardo** [comment]
Thank you very much @dsheng3 for your reply: I will try it out!

Best regards!
Bruno
